torchbenchmark/models/pytorch_stargan/main.py

```diff
@@ -0,0 +1,133 @@
+    parser.add_argument('--dataset', type=str, default='CelebA', choices=['CelebA', 'RaFD', 'Both'])
+    parser.add_argument('--batch_size', type=int, default=16, help='mini-batch size')
+    parser.add_argument('--num_iters', type=int, default=200000, help='number of total iterations for training D')
+    parser.add_argument('--num_iters_decay', type=int, default=100000, help='number of iterations for decaying lr')
+    parser.add_argument('--g_lr', type=float, default=0.0001, help='learning rate for G')
+    parser.add_argument('--d_lr', type=float, default=0.0001, help='learning rate for D')
+    parser.add_argument('--n_critic', type=int, default=5, help='number of D updates per each G update')
+    parser.add_argument('--beta1', type=float, default=0.5, help='beta1 for Adam optimizer')
+    parser.add_argument('--beta2', type=float, default=0.999, help='beta2 for Adam optimizer')
+    parser.add_argument('--resume_iters', type=int, default=None, help='resume training from this step')
+    parser.add_argument('--selected_attrs', '--list', nargs='+', help='selected attributes for the CelebA dataset',
+                        default=['Black_Hair', 'Blond_Hair', 'Brown_Hair', 'Male', 'Young'])
+
+    # Test configuration.
+    parser.add_argument('--test_iters', type=int, default=200000, help='test model from this step')
+
+    # Miscellaneous.
+    parser.add_argument('--num_workers', type=int, default=1)
+    parser.add_argument('--mode', type=str, default='train', choices=['train', 'test'])
+    parser.add_argument('--use_tensorboard', type=str2bool, default=True)
+    parser.add_argument('--device', type=str, default='cpu', choices=['cpu', 'cuda'])
+
+    # Directories.
+    parser.add_argument('--celeba_image_dir', type=str, default='data/celeba/images')
+    parser.add_argument('--attr_path', type=str, default='data/celeba/list_attr_celeba.txt')
+    parser.add_argument('--rafd_image_dir', type=str, default='data/RaFD/train')
+    parser.add_argument('--log_dir', type=str, default='stargan/logs')
+    parser.add_argument('--model_save_dir', type=str, default='stargan/models')
+    parser.add_argument('--sample_dir', type=str, default='stargan/samples')
+    parser.add_argument('--result_dir', type=str, default='stargan/results')
+
+    # Step size.
+    parser.add_argument('--log_step', type=int, default=10)
+    parser.add_argument('--sample_step', type=int, default=1000)
+    parser.add_argument('--model_save_step', type=int, default=10000)
+    parser.add_argument('--lr_update_step', type=int, default=1000)
+
+    # Extra arguments, not in the original stargan repo.
+    # --debug reference.out saves a file called "reference.out" with a result tensor
+    # from the last iteration of the model.
+    parser.add_argument('--debug', type=str, default='')
+    parser.add_argument('--deterministic', type=bool, default=False)
+    parser.add_argument('--should_script', type=bool, default=False)
+
+    config = parser.parse_args()
```

Comment:
this actually threw me a problem. we should figure out a way to test this in ci.  
when i ran locally `python test.py -k stargan` - the unittest parser correctly reads the -k argument and filters to only run your model.  but then the parser here trips over the -k that's still part of sys.argv and throws.

what i recommend doing is, 
L64: def parse_config(args=[])
L126: config = parser.parse_args(args)

that way by default all the defaults are used, but your utility can still be overridden if someone wanted to.